### PR TITLE
Bring in newpm (new pass manager) updates to master

### PR DESCRIPTION
**Pull request by pchintalapudi**
Ideally #46824 should just be flipping the define and confirming everything works; this brings in the other changes so far to master.

### CODE REVIEW

File: src/jitlayers.h [hunk 1]
```diff
@@ -25,6 +25,12 @@
 #include <stack>
 #include <queue>
 
+#define JL_USE_NEW_PM
```
Comment (pchintalapudi) [inline src/jitlayers.h:28]: Shouldn't forget to delete this
```suggestion
```

**Review by vchuravy — APPROVED** [review]

**Review by Keno — APPROVED** [review]

**Review by KristofferC — DISMISSED** [review]
https://github.com/JuliaLang/julia/pull/47038#discussion_r986420128 needs to be addressed.

**Comment by pchintalapudi** [comment]
`--time-passes` support is newly added now, which is rather useful to have.

**Review by vchuravy — APPROVED** [review]

**Comment by gbaraldi** [comment]
@pchintalapudi what is holding this up? I think LLVM 15 needs the new PM at least for sanitizers, but we may want to use it as an opportunity to turn it on everywhere

**Comment by pchintalapudi** [comment]
Mostly it's a broadcast regression, i need to fiddle with unswitch's knobs to make it go away.

**Comment by gbaraldi** [comment]
I guess we might want to fiddle with the pass ordering overall.

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/47038#issuecomment-1494574160) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/e767372_vs_46813d3/report.md).

**Review by vchuravy — APPROVED** [review]

**Comment by oscardssmith** [comment]
since this is passing tests, can it be merged?

**Pull request merged**